svg_test.go

```diff
@@ -0,0 +1,121 @@
+package jujusvg
+
+import (
+	"bytes"
+
+	gc "gopkg.in/check.v1"
+)
+
+type SVGSuite struct{}
+
+var _ = gc.Suite(&SVGSuite{})
+
+func (s *SVGSuite) TestProcessIcon(c *gc.C) {
+	tests := []struct {
+		about    string
+		icon     []byte
+		expected []byte
+		err      string
+	}{
+		{
+			about: "Nothing stripped",
+			icon: []byte(`
+				<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100">
+					<g id="foo"></g>
+				</svg>
+				`),
+			expected: []byte(`
+				<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100">
+					<g id="foo"></g>
+				</svg>`),
+		},
+		{
+			about: "ProcInst at start stripped",
+			icon: []byte(`
+				<?xml version="1.0"?>
+				<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100">
+					<g id="foo"></g>
+				</svg>
+				`),
+			expected: []byte(`
+				<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100">
+					<g id="foo"></g>
+				</svg>`),
+		},
+		{
+			about: "Directive at start stripped",
+			icon: []byte(`
+				<!DOCTYPE svg>
+				<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100">
+					<g id="foo"></g>
+				</svg>
+				`),
+			expected: []byte(`
+				<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100">
+					<g id="foo"></g>
+				</svg>`),
+		},
+		{
+			about: "ProcInst at end stripped",
+			icon: []byte(`
+				<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100">
+					<g id="foo"></g>
+				</svg>
+				<?xml foo="bar"?>
```

Comment:
@rogpeppe's right processing instructions are allowed after the main body, but <?xml... isn't a PI, it's an xmldecl. The parser treates them the same though, so you're not really testing anything by changing it.